src/utils/dates.js

```diff
@@ -40,3 +40,39 @@ export function getDaysBetweenDates(startDate, endDate) {
 		console.error('An error happened', error);
 	}
 }
+
+export function purchaseSchedule(dateLastPurchased, dateNextPurchased) {
+	const today = new Date();
+	const dateNextPurchased_formatted = dateNextPurchased.toDate();
+	let daysFromLastPurchase = 0; //in case when new item is added, by default daysFromLastPurchase == null, so assign a dummy value to proceed
+	if (dateLastPurchased) {
+		const dateLastPurchased_formatted = dateLastPurchased.toDate();
+		daysFromLastPurchase = getDaysBetweenDates(
+			dateLastPurchased_formatted,
+			today,
+		);
+	}
+
+	const daysTillNextPurchase = getDaysBetweenDates(
+		today,
+		dateNextPurchased_formatted,
+	);
+	let schedule = '';
+	if (dateHasPassed(dateNextPurchased) && daysFromLastPurchase < 60) {
+		schedule = 'overdue';
+	} else if (daysFromLastPurchase >= 60) {
+		schedule = 'inactive';
+	} else if (daysTillNextPurchase <= 7) {
+		schedule = 'soon';
+	} else if (daysTillNextPurchase > 7 && daysTillNextPurchase < 30) {
+		schedule = 'kind-of-soon';
```

Comment:
**Minor nitpick**: could we use "kind of soon" and "not soon" instead of including the dashes? From a user's perspective (and visually speaking), it would align better with the design and be a more user-friendly choice. However, I noticed that would change the span background color. To avoid that, my personal suggestion would be creating a mapping so we can display "not soon" when the backend value is "not-soon", something like this in `ListItem.jsx`:

```
const urgencyText = {
  'kind-of-soon': 'kind of soon',
  'not-soon': 'not soon',
};

// If a corresponding property exists in urgencyText, use it
// otherwise, keep the text as found in the backend.
const displayText = urgencyText[purchaseUrgency] || purchaseUrgency;

<span className={`urgency-tag ${purchaseUrgency}`}>
  {displayText}
</span>

``` 

And of course, there are probably different (and better) ways to do the same, but I wanted to make sure to give an example to facilitate (or at least give a starting point).

Reply:
Thank you so much Raquel, we were also thinking about the same thing and your solution works perfectly! We've incorporated it into our code

Reply:
@rdpfeifle This was a really clever suggestion, thank you!